app/main.py

```diff
@@ -1,3 +1,76 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = float(km)
+
+    def __str__(self) -> str:
+        return (
+            f"Distance: {int(self.km) if self.km.is_integer() else self.km} "
+            "kilometers."
+        )
```

Comment:
Do not use if condition here and in the `__repr__` as well
```suggestion
        return f"Distance: {self.km} kilometers."
```